File: typescript/docs_code_snippets/call-stream.ts
import { Opper } from "opperai";

const opper = new Opper();

for await (const chunk of opper.stream("creative_writer", {
  instructions: "Write a short story about a robot learning to paint.",
  input: { topic: "robot artist" },
})) {
  if (chunk.type === "content") {
    process.stdout.write(chunk.delta);
  }
  if (chunk.type === "done") {
    console.log();
  }
}

File: python/docs_code_snippets/call-stream.py
import sys
from opperai import Opper

opper = Opper()

# --- docs ---
for chunk in opper.stream(
    "creative_writer",
    instructions="Write a short story about a robot learning to paint.",
    input={"topic": "robot artist"},
):
    if chunk.type == "content":
        sys.stdout.write(chunk.delta)
        sys.stdout.flush()
    elif chunk.type == "done":
        print()
# --- /docs ---
